<!DOCTYPE html>
<html><head><title>Randy's Review Questions</title><link rel="stylesheet" href="../../css/style.css" type="text/css" /><script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script><script id="MathJax-script" async="" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script></head><h1>Randy's Review Questions</h1><h2>CS 167: Computational Biology</h2><p>Based on content taught by Donna Slonim in Spring 2022.</p><hr /><p class="question">1. The Needleman-Wunsch sequence alignment algorithm is an example of a(n)...</p><ul><input type="radio" name="q1" value="1" />iterative algorithm.<br /><input type="radio" name="q1" value="2" />greedy algorithm.<br /><input type="radio" name="q1" value="3" />dynamic programming algorithm.<br /><input type="radio" name="q1" value="4" />naive algorithm.<br /></ul><p id="q1correct" style="color:green;display:none;">Correct!</p><p id="q1incorrect" style="display:none;"><span style="color:purple;">Incorrect.</span> The correct answer was: dynamic programming algorithm.</p><p class="question">2. Consider the following scoring scheme for sequence alignment. This scoring scheme would be appropriate for...</p><pre>Match score = -3
Mismatch score = 1
Gap score = 2</pre><ul><input type="radio" name="q2" value="1" />global alignment.<br /><input type="radio" name="q2" value="2" />local alignment.<br /><input type="radio" name="q2" value="3" />All of the above<br /><input type="radio" name="q2" value="4" />None of the above<br /></ul><p id="q2correct" style="color:green;display:none;">Correct!</p><p id="q2incorrect" style="display:none;"><span style="color:purple;">Incorrect.</span> The correct answer was: global alignment.</p><p class="question">3. One parameter of the BLAST algorithm is a scoring threshold \(T\) (where "neighbor words" of a \(w\)-mer is defined as all sequences that align with the \(w\)-mer with a score of at least \(T\)). Lowering the \(T\) value would increase the _____, which would _____.</p><ul><input type="radio" name="q3" value="1" />false positive rate, sacrifice accuracy for speed<br /><input type="radio" name="q3" value="2" />false positive rate, sacrifice speed for accuracy<br /><input type="radio" name="q3" value="3" />false negative rate, sacrifice accuracy for speed<br /><input type="radio" name="q3" value="4" />false negative rate, sacrifice speed for accuracy<br /></ul><p id="q3correct" style="color:green;display:none;">Correct!</p><p id="q3incorrect" style="display:none;"><span style="color:purple;">Incorrect.</span> The correct answer was: false positive rate, sacrifice speed for accuracy</p><p class="question">4. When computing a best MSA using the bounded dynamic programming approach, which of the following would be an upper bound on the similarity score?</p><p class="question"><i>If there are multiple correct answers, select the one that would be the tightest upper bound.</i></p><ul><input type="radio" name="q4" value="1" />The optimum alignment score of the best-scoring pair in the alignment<br /><input type="radio" name="q4" value="2" />The optimum alignment score of the worst-scoring pair in the alignment<br /><input type="radio" name="q4" value="3" />The sum of the optimum alignment scores of all pairs in the alignment<br /><input type="radio" name="q4" value="4" />The sum-of-pairs score of any arbitrary MSA.<br /></ul><p id="q4correct" style="color:green;display:none;">Correct!</p><p id="q4incorrect" style="display:none;"><span style="color:purple;">Incorrect.</span> The correct answer was: The sum of the optimum alignment scores of all pairs in the alignment</p><p class="question">5. Consider the following three sequences. Suppose that we want to align them using the star alignment method. Assuming that all matches are treated the same, all mismatches are treated the same, and there are never any gaps, which sequence would be at the "center" of the star?</p><pre>Sequence 1: ATTCG
Sequence 2: ATACG
Sequence 3: ATTCC</pre><ul><input type="radio" name="q5" value="1" />Sequence 1<br /><input type="radio" name="q5" value="2" />Sequence 2<br /><input type="radio" name="q5" value="3" />Sequence 3<br /><input type="radio" name="q5" value="4" />There is a tie, so the sequence would be selected arbitrarily.<br /></ul><p id="q5correct" style="color:green;display:none;">Correct!</p><p id="q5incorrect" style="display:none;"><span style="color:purple;">Incorrect.</span> The correct answer was: Sequence 1</p><p class="question">6. Suppose that at position 2 in a local MSA, there are the following nucleotides: A, A, A, and A. Assuming pseudocounts of 1, what is the frequency of A at position 2?</p><ul><input type="radio" name="q6" value="1" />50%<br /><input type="radio" name="q6" value="2" />62.50%<br /><input type="radio" name="q6" value="3" />87.50%<br /><input type="radio" name="q6" value="4" />100%<br /></ul><p id="q6correct" style="color:green;display:none;">Correct!</p><p id="q6incorrect" style="display:none;"><span style="color:purple;">Incorrect.</span> The correct answer was: 62.50%</p><p class="question">7. The compressive BLAST algorithm involves _____ preprocessing and _____ search.</p><ul><input type="radio" name="q7" value="1" />sublinear, sublinear<br /><input type="radio" name="q7" value="2" />linear, sublinear<br /><input type="radio" name="q7" value="3" />polynomial, linear<br /><input type="radio" name="q7" value="4" />exponential, linear<br /></ul><p id="q7correct" style="color:green;display:none;">Correct!</p><p id="q7incorrect" style="display:none;"><span style="color:purple;">Incorrect.</span> The correct answer was: linear, sublinear</p><p class="question">8. Suppose that there exists a branching node in a de Bruijn graph of \(k\)-mers. What does this imply?</p><ul><input type="radio" name="q8" value="1" />There exists an Eulerian path in the equivalent graph of \((k-1)\)-mers.<br /><input type="radio" name="q8" value="2" />There does not exist an Eulerian path in the equivalent graph of \((k-1)\)-mers.<br /><input type="radio" name="q8" value="3" />There exists an Eulerian path in the equivalent graph of \((k+1)\)-mers.<br /><input type="radio" name="q8" value="4" />There does not exist an Eulerian path in the equivalent graph of \((k+1)\)-mers.<br /></ul><p id="q8correct" style="color:green;display:none;">Correct!</p><p id="q8incorrect" style="display:none;"><span style="color:purple;">Incorrect.</span> The correct answer was: There does not exist an Eulerian path in the equivalent graph of \((k-1)\)-mers.</p><p class="question">9. Consider the following overlap graph. What is one possible result of executing greedy assembly on that graph?</p><img src="testq1.PNG" alt="testq1.PNG" /><ul><input type="radio" name="q9" value="1" /><code>TACAGTTCAGA</code><br /><input type="radio" name="q9" value="2" /><code>TACAGAGTTCA</code><br /><input type="radio" name="q9" value="3" /><code>TACAGGTTCACAGA</code><br /><input type="radio" name="q9" value="4" /><code>TACAGCAGAGTTCA</code><br /></ul><p id="q9correct" style="color:green;display:none;">Correct!</p><p id="q9incorrect" style="display:none;"><span style="color:purple;">Incorrect.</span> The correct answer was: <code>TACAGAGTTCA</code></p><p class="question">10. Organisms that overall have a high GC content will have the most significant bias towards G and C in the _____ base of a codon.</p><ul><input type="radio" name="q10" value="1" />first<br /><input type="radio" name="q10" value="2" />second<br /><input type="radio" name="q10" value="3" />third<br /><input type="radio" name="q10" value="4" />None of the above<br /></ul><p id="q10correct" style="color:green;display:none;">Correct!</p><p id="q10incorrect" style="display:none;"><span style="color:purple;">Incorrect.</span> The correct answer was: third</p><p class="question">11. Which of the following algorithms finds the most likely state sequence when given a sequence of observable symbols and the specification of a Hidden Markov Model?</p><ul><input type="radio" name="q11" value="1" />Forward algorithm<br /><input type="radio" name="q11" value="2" />Viterbi algorithm<br /><input type="radio" name="q11" value="3" />Maximum Likelihood Estimation<br /><input type="radio" name="q11" value="4" />Baum-Welch algorithm<br /></ul><p id="q11correct" style="color:green;display:none;">Correct!</p><p id="q11incorrect" style="display:none;"><span style="color:purple;">Incorrect.</span> The correct answer was: Viterbi algorithm</p><p class="question">12. Suppose that in a given experiment, three people with sickle cell anemia had the following levels of expression of Gene 1: 7, 9, 11. Further suppose that three people without the disease had the following levels of expression of the same gene: 2, 4, 3. On average, there was a(n) _____ in expression of Gene 1 for those with sickle-cell anemia.</p><ul><input type="radio" name="q12" value="1" />3-fold increase<br /><input type="radio" name="q12" value="2" />6-fold increase<br /><input type="radio" name="q12" value="3" />3-fold decrease<br /><input type="radio" name="q12" value="4" />6-fold decrease<br /></ul><p id="q12correct" style="color:green;display:none;">Correct!</p><p id="q12incorrect" style="display:none;"><span style="color:purple;">Incorrect.</span> The correct answer was: 3-fold increase</p><p class="question">13. The molecular clock hypothesis holds for _____ trees, which satisfy the _____ condition.</p><ul><input type="radio" name="q13" value="1" />additive, 3-point<br /><input type="radio" name="q13" value="2" />additive, 4-point<br /><input type="radio" name="q13" value="3" />ultrametric, 3-point<br /><input type="radio" name="q13" value="4" />ultrametric, 4-point<br /></ul><p id="q13correct" style="color:green;display:none;">Correct!</p><p id="q13incorrect" style="display:none;"><span style="color:purple;">Incorrect.</span> The correct answer was: ultrametric, 3-point</p><p class="question">14. The problem of finding the most parsimonious tree when given an MSA can be solved in... (select the one that is most efficient)</p><ul><input type="radio" name="q14" value="1" />sublinear time.<br /><input type="radio" name="q14" value="2" />linear time.<br /><input type="radio" name="q14" value="3" />polynomial time.<br /><input type="radio" name="q14" value="4" />None of the above<br /></ul><p id="q14correct" style="color:green;display:none;">Correct!</p><p id="q14incorrect" style="display:none;"><span style="color:purple;">Incorrect.</span> The correct answer was: None of the above</p><p class="question">15. Anomaly detection is a method of classification that seeks to classify data points as "normal" or "anomalous." Such a classifier is trained with...</p><ul><input type="radio" name="q15" value="1" />normal data points only.<br /><input type="radio" name="q15" value="2" />anomalous data points only.<br /><input type="radio" name="q15" value="3" />labelled examples of normal and anomalous data points.<br /><input type="radio" name="q15" value="4" />unlabelled examples of normal and anomalous data points.<br /></ul><p id="q15correct" style="color:green;display:none;">Correct!</p><p id="q15incorrect" style="display:none;"><span style="color:purple;">Incorrect.</span> The correct answer was: normal data points only.</p><h3 id="score" style="color:blue;display:none;"></h3><input type="button" onclick="checkquizans()" value="Check Answers" /><br /><hr /><a href="../../index.html">Return to Home</a><script src="../../scripts/checkans.js"></script><script>function checkquizans() { answerKey = [["3"], ["1"], ["2"], ["3"], ["1"], ["2"], ["2"], ["2"], ["2"], ["3"], ["2"], ["1"], ["3"], ["4"], ["1"]]; checkans(answerKey); }</script><noscript><p>Looks like JavaScript is disabled! The answer checker will not work without JavaScript enabled.</p></noscript></html>